### PR TITLE
feat: fig autocomplete feature

### LINKED ISSUE #1359: generate fig autocompletion
**Description**

We (w/@srenatus @anderseknert) did a [similar effort](https://github.com/withfig/autocomplete/pull/854) in OPA to add autocompletion of opa CLI in by [fig](https://github.com/withfig/autocomplete). So, let's do this for cosign too.

image (17)

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

#### Summary

will generate autocompletion typescript code of cosign for fig

#### Ticket Link

Fixes #1359 

#### Release Note

```release-note
feat: fig autocomplete feature
```

### CODE REVIEW

**Comment by developer-guy** [comment]
kindly ping 🙋🏻‍♂️

**Comment by dlorenc** [comment]
Tried to re-run the CI, something failed in the go-releaser snapshot...

**Comment by dlorenc** [comment]
Still failing, maybe try a rebase?

```shell
/bin/sh: 1: goimports: not found
LDFLAGS="-X github.com/sigstore/cosign/pkg/version.GitVersion=21ca2f4 -X github.com/sigstore/cosign/pkg/version.gitCommit=21ca2f407f5261cee7a10089f9de0f17d5e259ec -X github.com/sigstore/cosign/pkg/version.gitTreeState="clean" -X github.com/sigstore/cosign/pkg/version.buildDate='2022-02-01T07:07:41Z'" goreleaser release --skip-sign --skip-publish --snapshot --rm-dist
   • releasing...             
   • loading config file       file=.goreleaser.yml
   • loading environment variables
   • getting and validating git state
      • ignoring errors because this is a snapshot error=git doesn't contain any tags. Either add a tag or use --snapshot
      • building...               commit=21ca2f407f5261cee7a10089f9de0f17d5e259ec latest tag=v0.0.0
      • pipe skipped              error=disabled during snapshot mode
   • parsing tag              
   • running before hooks     
      • running                   hook=go mod tidy
      • running                   hook=/bin/bash -c 'if [ -n "$(git --no-pager diff --exit-code go.mod go.sum)" ]; then exit 1; fi'
   ⨯ release failed after 43.37s error=hook failed: /bin/bash -c 'if [ -n "$(git --no-pager diff --exit-code go.mod go.sum)" ]; then exit 1; fi': exit status 1; output: 
make: *** [release/release.mk:51: snapshot] Error 1
Error: Process completed with exit code 2.
```

**Comment by developer-guy** [comment]
cc: @cpanato, is that something related to an image ghcr.io/gythialy/golang-cross:v1.17.6-3@sha256:312ac8449408302e5fdde452578607cff075bc80052f4526254cd25fa96ce9e0?

**Comment by cpanato** [comment]
> cc: @cpanato, is that something related to an image `ghcr.io/gythialy/golang-cross:v1.17.6-3@sha256:312ac8449408302e5fdde452578607cff075bc80052f4526254cd25fa96ce9e0`? Screen Shot 2022-02-09 at 12 00 06

can you try to run go mod tidy and push the changes? if you ran and there is no change I will take a look

**Comment by developer-guy** [comment]
> > cc: @cpanato, is that something related to an image `ghcr.io/gythialy/golang-cross:v1.17.6-3@sha256:312ac8449408302e5fdde452578607cff075bc80052f4526254cd25fa96ce9e0`? Screen Shot 2022-02-09 at 12 00 06
> 
> can you try to run go mod tidy and push the changes? if you ran and there is no change I will take a look

you are like a hero 🦸

**Comment by cpanato** [comment]
> but same error :/

i will checkout your pr, give me some time

**Comment by cpanato** [comment]
running the job locally it works

**Comment by cpanato** [comment]
in ci it is complaing about

```
--- a/go.mod
+++ b/go.mod
@@ -58,6 +58,7 @@ require (
 	go.opentelemetry.io/contrib v1.3.0 // indirect
 	go.opentelemetry.io/proto/otlp v0.12.0 // indirect
 	golang.org/x/net v0.0.0-20220121210[141](https://github.com/sigstore/cosign/runs/5127527740?check_suite_focus=true#step:5:141)-e204ce36a2ba // indirect
+	golang.org/x/tools v0.1.9 // indirect
 	k8s.io/kube-openapi v0.0.0-20220124234850-424119656bbf // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
 )
diff --git a/go.sum b/go.sum
index be66de7..42b62af 100644
--- a/go.sum
+++ b/go.sum
@@ -2494,8 +2494,9 @@ golang.org/x/tools v0.1.3/go.mod h1:o0xws9oXOQQZyjljx8fwUC0k7L1pTE6eaCbjGeHmOkk=
 golang.org/x/tools v0.1.4/go.mod h1:o0xws9oXOQQZyjljx8fwUC0k7L1pTE6eaCbjGeHmOkk=
 golang.org/x/tools v0.1.5/go.mod h1:o0xws9oXOQQZyjljx8fwUC0k7L1pTE6eaCbjGeHmOkk=
 golang.org/x/tools v0.1.7/go.mod h1:LGqMHiF4EqQNHR1JncWGqT5BVaXmza+X+BDGol+dOxo=
-golang.org/x/tools v0.1.8 h1:P1HhGGuLW4aAclzjtmJdf0mJOjVUZUzOTqkAkWL+l6w=
 golang.org/x/tools v0.1.8/go.mod h1:nABZi5QlRsZVlzPpHl034qft6wpY4eDcsTt5AaioBiU=
+golang.org/x/tools v0.1.9 h1:j9KsMiaP1c3B0OTQGth0/k+miLGTgLsAFUCrF2vLcF8=
+golang.org/x/tools v0.1.9/go.mod h1:nABZi5QlRsZVlzPpHl034qft6wpY4eDcsTt5AaioBiU=
 golang.org/x/xerrors v0.0.0-20190717185122-a985d3407aa7/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
 golang.org/x/xerrors v0.0.0-20191011141410-1b5[146](https://github.com/sigstore/cosign/runs/5127527740?check_suite_focus=true#step:5:146)add898/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
 golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
```

**Comment by cpanato** [comment]
also made a pr to add goimports in the cross builder image: https://github.com/gythialy/golang-cross/pull/87

**Comment by cpanato** [comment]
and i think i discover why, one sec

File: .github/workflows/validate-release.yml [hunk 1]
```diff
@@ -45,6 +45,9 @@ jobs:
     steps:
       - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 #v2.4.0
 
+      - name: Install goimports
+        run: go get golang.org/x/tools/cmd/goimports
```
Comment (cpanato) [inline .github/workflows/validate-release.yml:49]: ok, now i see this:
if you do this you add this tool as a dependency in the project and then the check we do for the go modules fail.

please remove this and the job will pass

**Review by cpanato — CHANGES_REQUESTED** [review]

File: .github/workflows/validate-release.yml (hunk 1 above)
Comment (developer-guy) [reply .github/workflows/validate-release.yml]: thank you, I've removed these lines.

**Comment by developer-guy** [comment]
thank you so much for helping me @cpanato, you're the best

**Review by cpanato — APPROVED** [review]

**Pull request merged**